packages/hyperliquid-composer/contracts/extensions/PreFundedFeeAbstraction.sol

```diff
@@ -0,0 +1,235 @@
+     * @dev Transfers USDC tokens from the composer's HyperCore balance to the specified address
+     * @dev Can only be called by the recovery address
+     * @param _coreAmount Amount of USDC tokens to retrieve in HyperCore decimals, or FULL_TRANSFER for all
+     * @param _to Destination address to receive the retrieved USDC tokens
+     */
+    function retrieveCoreUSDC(uint64 _coreAmount, address _to) public virtual override onlyRecoveryAddress {
+        uint64 maxTransferAmt = _getMaxTransferAmount(USDC_CORE_INDEX, _coreAmount);
+
+        uint64 minComposerBalance = _coreAmount == FULL_TRANSFER
+            ? accruedActivationFee - MIN_USDC_PRE_FUND_AMOUNT()
+            : _coreAmount;
+
+        if (maxTransferAmt > minComposerBalance) {
+            revert MaxRetrieveAmountExceeded(minComposerBalance, maxTransferAmt);
+        }
+
+        _submitCoreWriterTransfer(_to, USDC_CORE_INDEX, maxTransferAmt);
+        emit Retrieved(USDC_CORE_INDEX, maxTransferAmt, _to);
+    }
+
+    /**
+     * @notice Gets current spot price from HyperCore precompile
+     * @return Raw spot price with protocol-specific decimal encoding
+     */
+    function _spotPx() internal view returns (uint64) {
+        bool success;
+        bytes memory result;
+        (success, result) = SPOT_PX_PRECOMPILE_ADDRESS.staticcall(abi.encode(SPOT_ID));
+        require(success, "SpotPx precompile call failed");
+        return abi.decode(result, (uint64));
+    }
+
+    /**
+     * @notice Gets spot pair info from HyperCore precompile
+     * @param _spot Spot pair ID to query
+     * @return SpotInfo containing pair name and token indices
+     */
+    function _spotInfo(uint64 _spot) internal view returns (SpotInfo memory) {
+        bool success;
+        bytes memory result;
+        (success, result) = SPOT_INFO_PRECOMPILE_ADDRESS.staticcall(abi.encode(_spot));
+        require(success, "SpotInfo precompile call failed");
+        return abi.decode(result, (SpotInfo));
+    }
+
+    /**
+     * @notice Gets token metadata from HyperCore precompile
+     * @param _coreSpotIndex Core token index to query
+     * @return TokenInfo with token metadata and decimal configurations
+     */
+    function _tokenInfo(uint64 _coreSpotIndex) internal view returns (TokenInfo memory) {
+        bool success;
+        bytes memory result;
+        (success, result) = TOKEN_INFO_PRECOMPILE_ADDRESS.staticcall(abi.encode(_coreSpotIndex));
+        require(success, "TokenInfo precompile call failed");
+        return abi.decode(result, (TokenInfo));
+    }
+
+    /**
+     * @notice Returns the minimum activation cost threshold in USD terms
+     * @dev Can be overridden by implementing contracts to adjust minimum fee requirements
+     * @return The minimum USD amount required for activation fees
+     */
+    function MIN_USDC_PRE_FUND_AMOUNT() public pure virtual returns (uint8) {
+        return DEFAULT_MIN_USDC_PRE_FUND_AMOUNT;
```

Comment:
Can we remove this function and rename `DEFAULT_MIN_USDC_PRE_FUND_AMOUNT` to `MIN_USDC_PRE_FUND_AMOUNT`?